storage/store_test.go

```diff
@@ -1550,3 +1551,19 @@ func TestMaybeRemove(t *testing.T) {
 		t.Errorf("Unexpected removed range %v", removedRng)
 	}
 }
+
+func TestErrWithIndex(t *testing.T) {
+	defer leaktest.AfterTest(t)
+	ind := int32(99)
+	err := error(&errWithIndex{err: errors.New("foo"), index: ind})
+	i1, ok1 := err.(proto.IndexedError).ErrorIndex()
+	pErr := &proto.Error{}
+	pErr.SetResponseGoError(err)
+	if pErr.GetIndex().GetIndex() != ind {
+		log.Warningf("proto.Error lost index information")
```

Comment:
this just makes sure that `SetResponseGoError()` will set the `Index` in the `*proto.Error` message. If `err` implemented `ErrorDetail` (doesn't now, but could in the future), the next test using `GoError()` below would get the information from the original error, but it should do the same in either case.